### ISSUE TITLE
💅 CSS Linter cannot handle partial data

### ISSUE THREAD

**Issue by samcragg** [biomejs/biome#10905]
### Environment information

I'm using the VS Code extension and have noticed in 2.5.2 and 2.5.3 (I've not been able to reproduce it in 2.5.0 or 2.5.1) that when editing CSS files I get constant errors popping up (not inline errors in the document, but popups that the service has failed). I've created a minimal example, to reproduce, just change the `example.css` (as in, it's when typing in the document things go wrong)

<img width="463" height="99" alt="Image" src="https://github.com/user-attachments/assets/2608f6e1-b457-4d0a-bbc5-9afac27cfa12" />

```
2026-07-12 07:05:21.033 [info] [Info  - 7:05:21 AM] Server initialized with PID: 25820
2026-07-12 07:05:40.293 [info] Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates\biome_rowan\src\ast\mod.rs:891:32
Thread Name: tokio-rt-worker
Message: can't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace


2026-07-12 07:06:54.152 [info] [Error - 7:06:54 AM] Request textDocument/codeAction failed.
2026-07-12 07:06:54.152 [info]   Message: panicked at crates\biome_rowan\src\ast\mod.rs:891:32:
can't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29
Backtrace: disabled backtrace
  Code: -32603 
PanicError { info: "panicked at crates\\biome_rowan\\src\\ast\\mod.rs:891:32:\ncan't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29", backtrace: Some(<disabled>) }
2026-07-12 07:07:20.985 [info] [Error - 7:07:20 AM] Request textDocument/codeAction failed.
2026-07-12 07:07:20.985 [info]   Message: panicked at crates\biome_rowan\src\ast\mod.rs:891:32:
can't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29
Backtrace: disabled backtrace
  Code: -32603 
PanicError { info: "panicked at crates\\biome_rowan\\src\\ast\\mod.rs:891:32:\ncan't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29", backtrace: Some(<disabled>) }
2026-07-12 07:07:21.079 [info] [Error - 7:07:21 AM] Request textDocument/codeAction failed.
2026-07-12 07:07:21.079 [info]   Message: panicked at crates\biome_rowan\src\ast\mod.rs:891:32:
can't resolve AstPtr { raw: SyntaxNodePtr { kind: CSS_GENERIC_COMPONENT_VALUE_LIST, range: 24..24 } } with CSS_ROOT@0..29
Backtrace: disabled backtrace
  Code: -32603 
```

### Rule name

Recommended CSS rules

### Playground link

https://github.com/samcragg/biome_css_bug

### Expected result

The language server should be able to handle partial CSS files whilst typing

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by suryarupi** [comment]
Hi @ematipico, I'd like to investigate this issue and work on a fix if it is still available. I was able to understand the reported crash and will try reproducing it locally before looking into the AST pointer resolution issue.
